superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -501,22 +529,79 @@ export default class ResultSet extends React.PureComponent<
     return <div />;
   }
 
+  onAlertClose = () => {
+    this.setState({ alertIsOpen: false });
+  };
+
   renderRowsReturned() {
-    const { results, rows, queryLimit } = this.props.query;
-    const limitReached = results?.displayLimitReached;
+    const { results, rows, queryLimit, limitingFactor } = this.props.query;
+    let limitMessage;
+    const isAdmin = this.props.user?.roles.hasOwnProperty('Admin');
```

Comment:
why `hasOwnProperty` here?

Reply:
or you could also Object.keys(this.props.user?.roles).includes("Admin")

Reply:
it's not that `hasOwnProperty` doesn't work, it's just that it was mainly used to make sure that the object isn't inheriting the property from something up the prototype chain, and so I was trying to think about why that was an issue. I think the other methods are more common nowadays.